### PR TITLE
AI Chat error states

### PR DESCRIPTION
**Pull request by vadimstruts**
Implemented two error states:
- "You have reached the current rate limit for Leo. Please try again later" - rate limit has been reached
- "There was a network issue connecting to Leo, check your connection and try again" - for all other cases

We show the error description and Retry button, according to the Figma (https://www.figma.com/file/h7fKSJF2P556bZWfyLTfwD/Brave-AI?type=design&node-id=1520%3A30055&mode=design&t=BIJ98u8xeRJPSD06-1)

If `Retry` button pressed, the message  is retried/re-sent and successful response is printed instead of an error.

Short video: 
https://github.com/brave/brave-core/assets/118171981/54b9c3f5-71cb-4b0d-b885-52d3cf5e58e3

Resolves https://github.com/brave/brave-browser/issues/31391

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/resources/ai_chat_ui_strings.grdp [hunk 1]
```diff
@@ -28,7 +28,10 @@
     Ask Brave AI
   </message>
   <message name="IDS_CHAT_UI_API_ERROR" desc="Describes an error occured submitting the API request">
-    There was an error processing your message, please try again.
+    There was a network issue connecting to Leo, check your connection and try again
+  </message>
+  <message name="IDS_CHAT_UI_API_TOO_MANY_REQUESTS_ERROR" desc="Describes an error occured when max count of API requests exceeded">
+    You have reached the current rate limit for Leo. Please try again later
```
Comment (mkarolin) [inline components/resources/ai_chat_ui_strings.grdp:34]: @rmcfadden3, is this too technical?

Comment (rmcfadden3) [reply components/resources/ai_chat_ui_strings.grdp:34]: @mkarolin — oof. sorry for missing your ping.

Yeah, this error message is too technical. I checked with @rebron, and it sounds like rate limiting here basically means the number of questions you've asked in a given time span (i.e one day). So, given that, I'd suggest editing to something like:

`You've reached the maximum number of questions for Leo. Please try again tomorrow.`

… or something along those lines.

Oh, and I assume the rate limit is increased (or even non-existent) for Leo Premium? If yes, we could amend the second sentence to something like `Please try again tomorrow, or upgrade to Leo Premium.` (if the UI allows for that many characters).

Comment (vadimstruts) [reply components/resources/ai_chat_ui_strings.grdp]: fixed: https://github.com/brave/brave-core/pull/19318/commits/458ba9c9e82f11cec4bf347efc6101f0c04df9f7#diff-dbd0ebfba47f26a3cbccd8f40b0660933c0040d1cbea77d2ac4ffe4e2d56f853R34

**Review by nullhook — CHANGES_REQUESTED** [review]

File: components/resources/ai_chat_ui_strings.grdp (hunk 1 above)
Comment (rebron) [reply components/resources/ai_chat_ui_strings.grdp]: cc:@nullhook Let's remove the ", or upgrade to Leo Premium." line.

**Comment by petemill** [comment]
As discussed, @nullhook ended up continuing this work in https://github.com/brave/brave-core/pull/19804 so we can close this one. Thanks for the effort!

**Comment by vadimstruts** [comment]
Closed due to https://github.com/brave/brave-core/pull/19804

**Pull request closed without merging**